src/index.js

```diff
@@ -187,6 +187,8 @@ class ServerlessSnsToSqsEvents {
 			return this.getLogicalId(snsName, "Topic");
 		} else if (snsArn.Ref) {
 			return snsArn.Ref;
+		} else if (snsArn["Fn::ImportValue"]) {
+			return this.getLogicalId(snsArn["Fn::ImportValue"], "Topic");
```

Comment:
have you checked what does this produce?

Reply:
looks like it's used to create the LogicalId for the SNS subscription, it might be quite long but should be OK I imagine, but would be useful to see an example of what it looks like.

Reply:
If I log the resources created at the end of the test I added, we get:

```json
{
  "HelloLambdaFunctionMyExportedTopicTopicToMyQueueSubscription": {
    "Type": "AWS::SNS::Subscription",
    "Properties": {
      "Protocol": "sqs",
      "Endpoint": {
        "Fn::GetAtt": [
          "MyQueue",
          "Arn"
        ]
      },
      "TopicArn": {
        "Fn::ImportValue": "MyExportedTopic"
      }
    }
  },
  "HelloLambdaFunctionMyExportedTopicTopicToMyQueueQueuePolicy": {
    "Type": "AWS::SQS::QueuePolicy",
    "Properties": {
      "Queues": [
        {
          "Ref": "MyQueue"
        }
      ],
      "PolicyDocument": {
        "Version": "2012-10-17",
        "Statement": {
          "Effect": "Allow",
          "Principal": "*",
          "Action": "SQS:SendMessage",
          "Resource": {
            "Fn::GetAtt": [
              "MyQueue",
              "Arn"
            ]
          },
          "Condition": {
            "ArnEquals": {
              "aws:SourceArn": {
                "Fn::ImportValue": "MyExportedTopic"
              }
            }
          }
        }
      }
    }
  }
}
```

If I update the import name to "myservice-MyExportedTopic", we get:


```json
{
  "HelloLambdaFunctionMyserviceDashMyExportedTopicTopicToMyQueueSubscription": {
    "Type": "AWS::SNS::Subscription",
    "Properties": {
      "Protocol": "sqs",
      "Endpoint": {
        "Fn::GetAtt": [
          "MyQueue",
          "Arn"
        ]
      },
      "TopicArn": {
        "Fn::ImportValue": "myservice-MyExportedTopic"
      }
    }
  },
  "HelloLambdaFunctionMyserviceDashMyExportedTopicTopicToMyQueueQueuePolicy": {
    "Type": "AWS::SQS::QueuePolicy",
    "Properties": {
      "Queues": [
        {
          "Ref": "MyQueue"
        }
      ],
      "PolicyDocument": {
        "Version": "2012-10-17",
        "Statement": {
          "Effect": "Allow",
          "Principal": "*",
          "Action": "SQS:SendMessage",
          "Resource": {
            "Fn::GetAtt": [
              "MyQueue",
              "Arn"
            ]
          },
          "Condition": {
            "ArnEquals": {
              "aws:SourceArn": {
                "Fn::ImportValue": "myservice-MyExportedTopic"
              }
            }
          }
        }
      }
    }
  }
}
```

Does that seem ok? This output would work for my use case

Reply:
thanks for checking, yeah, that looks fine, hopefully it fixes the problem for you!

Reply:
Hooray! Thanks!